pandas/core/series.py

```diff
@@ -1199,9 +1199,8 @@ def reset_index(self, level=None, drop=False, name=None, inplace=False):
                 if not isinstance(level, (tuple, list)):
                     level = [level]
                 level = [self.index._get_level_number(lev) for lev in level]
-                if isinstance(self.index, MultiIndex):
-                    if len(level) < self.index.nlevels:
-                        new_index = self.index.droplevel(level)
```

Comment:
are there more places u can simplify? maybe in core/indexing.py ?

Reply:
Found only another one. The other uses of droplevel do need to treat differently flat indexes.